include/ck/utility/reduction_operator.hpp

```diff
@@ -59,6 +59,14 @@ struct Add
         a = a + b;
     }
 
+    __host__ __device__ inline constexpr void operator()(int8_t& a, int8_t b) const
+    {
+        int32_t a_ = type_convert<int32_t>(a);
+        int32_t b_ = type_convert<int32_t>(b);
+
+        a = type_convert<int8_t>(a_ + b_);
+    }
```

Comment:
Why we need this? int8 is a standard type... I don't think we should add two int8 values together in such a way. Rather we use ComputeDataType with wider range to make sure to not overflow and in the end cast back to output data type.

Reply:
Right, my mistake. Changed